bodhi/server/validators.py

```diff
@@ -149,6 +148,66 @@ def validate_csrf_token(node, value):
         raise colander.Invalid(node, csrf_error_message)
 
 
+def cache_tags(request, build):
+    """
+    Cache the tags for a koji build.
+
+    Args:
+        request (pyramid.util.Request): The current request.
+        build (basestring): The NVR of the build to cache.
+    Returns:
+        The list of tags or None.
+    """
+    if build in request.buildinfo and 'tags' in request.buildinfo[build]:
+        return request.buildinfo[build]['tags']
+    tags = None
+    try:
+        tags = [tag['name'] for tag in request.koji.listTags(build)]
+        if len(tags) == 0:
+            request.errors.add('body', 'builds',
+                               'Cannot find any tags associated with build: %s' % build)
+    except koji.GenericError:
+        request.errors.add('body', 'builds',
+                           'Invalid koji build: %s' % build)
+    # This might end up setting tags to None. That is expected, and indicates it failed.
+    request.buildinfo[build]['tags'] = tags
+    return tags
+
+
+def cache_release(request, build):
+    """
+    Cache the builds release from the request.
+
+    Args:
+        request (pyramid.util.Request): The current request.
+        build (basestring): The NVR of the build to cache.
+    Returns:
+        The release object or None.
```

Comment:
Release or None: The release object, or None if no Release can be matched to the tags associated with the request.